### ISSUE TITLE
Padding API conventions

### ISSUE THREAD

**Issue by orsharir** [HUJI-Deep/simnets-tf#14]
Notice that in Tensorflow the padding api is specified in uppercase, i.e. ``'SAME'`` and  ``'VALID'``, while in Keras it is in lowercase, i.e. ``'same'`` and  ``'valid'``. Our API should adhere to this convention. You could probably use [this Keras's method](https://github.com/fchollet/keras/blob/master/keras/backend/tensorflow_backend.py#L3008) to handle this change (or simply write your own).
